File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/RobotLostMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class RobotLostMessage extends Message {	
	public static int MESSAGE_ID				= 204;
	
	public RobotLostMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public RobotLostMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return false;
	}	
}

File: maven-root/mtsa/src/main/java/RobotProtocol/src/robotProtocol/messages/RobotLostMessage.cpp
/*
 * RobotLostMessage.cpp
 *
 *  Created on: Apr 16, 2013
 *      Author: mariano
 */

#include "RobotLostMessage.h"

namespace RobotProtocol {

RobotLostMessage RobotLostMessage::staticMessage = RobotLostMessage(0,0);

Message *RobotLostMessage::deserialize(char *serializedMessage){
	RobotLostMessage::staticMessage.initialize(serializedMessage);
	return &(RobotLostMessage::staticMessage);
}

RobotLostMessage::RobotLostMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}

RobotLostMessage::RobotLostMessage(char *serializedMessage){
	initialize(serializedMessage);
}

RobotLostMessage::~RobotLostMessage() {
}

int RobotLostMessage::getMessageLength(){
	return Message::getMessageLength();
}

void RobotLostMessage::readData(char **serializedMessageIndex){}

void RobotLostMessage::writeData(char **serializedMessageIndex){}


} /* namespace RobotProtocol */
